awesome_sso/store/minio.py

```diff
@@ -40,13 +40,32 @@ def __init__(
         else:
             self.logger.info("bucket '%s' exists", self.bucket)
 
-    def fput(self, name: str, file_path: str):
+    def list_buckets(self):
+        """List information of all accessible buckets with text."""
+        return [x.name for x in self.client.list_buckets()]
+
+    def list_objects(self, prefix: str = None, recursive: bool = False):
+        """Lists object information of a bucket with text."""
+        return [
+            x.object_name
+            for x in self.client.list_objects(
+                self.bucket, prefix=prefix, recursive=recursive
+            )
+        ]
+
+    def fput(self, name: str, file_path: str, exclude_files: List[str] = []):
+        """Uploads data from a file/folder to an object in a bucket."""
         if path.isdir(file_path):
             for local_file in glob.glob(file_path + "/**"):
+                file_name = local_file[1 + len(file_path) :]
```

Comment:
this hard-coded way to extract filename would fail, if we have more then one layer of directory. User
```
from pathlib import Path

file_name = Path(file_path).name
if file_name in exclude_files:
    continue
```